noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/StreamFeedSocketAdapterClient.java

```diff
@@ -0,0 +1,64 @@
+package edu.uci.ics.cloudberry.noah.feed;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+
+public class StreamFeedSocketAdapterClient {
```

Comment:
Those `XAdapters` are sharing the same `initialize`, `finalize`, `ingest` interfaces. Could you restructure the code to let them inherit the basic one and sharing the common codes?